widget/view.controller.js

```diff
@@ -6,85 +6,404 @@
-                        $scope.picklistField = entity.fields.threatHuntTools;
-                        $scope.processingPicklist = false;
+            var queryBody = {
+                "logic": "AND",
+                "filters": [
+                    {
+                        "type": "primitive",
+                        "field": "key",
+                        "value": "outbreak-threat-hunt-tools",
+                        "operator": "eq",
+                        "_operator": "eq"
                     }
+                ]
+            };
+            var queryString = {
+                $limit: ALL_RECORDS_SIZE
+            };
+            $resource(API.QUERY + 'keys').save(queryString, queryBody).$promise.then(function (response) {
+                if (response['hydra:member'] && response['hydra:member'].length > 0) {
+                    $scope.processingPicklist = false;
+                    $scope.huntToolsMapping = response['hydra:member'][0].jSONValue;
+                    $scope.threatHuntTools = Object.keys($scope.huntToolsMapping);
+                    WizardHandler.wizard('OutbreaksolutionpackWizard').next();
+                }
+                else {
+                    toaster.error({
+                        body: 'Threat Hunt Tool is not found in Key-Store'
+                    });
+                    return;
                 }
             });
 
-            WizardHandler.wizard('solutionpackWizard').next();
         }
 
-        function moveTktNext() {
-            $scope.ticketingPicklist = true;
-            var entity = new Entity('outbreak_alerts');
-            entity.loadFields().then(function () {
-                for (var key in entity.fields) {
-                    if (entity.fields[key].type === 'picklist' && key === 'ticketingTools') {
-                        $scope.tktPicklistField = entity.fields.ticketingTools;
-                        $scope.ticketingPicklist = false;
+        function _fetchConnectorConfig(connectorName) {
+            var queryBody = {
+                "logic": "AND",
+                "filters": [
+                    {
+                        "field": "name",
+                        "operator": "in",
+                        "value": connectorName
                     }
+                ]
+            };
+            $resource(API.QUERY + 'solutionpacks').save({ $limit: ALL_RECORDS_SIZE }, queryBody).$promise.then(function (response) {
+                if (response['hydra:member'] || response['hydra:member'].length > 0) {
+                    var huntToolDetails = _.map(response['hydra:member'], obj => _.pick(obj, ['name', 'label', 'version', 'uuid']));
+                    _loadConnectorDetails(huntToolDetails[0].name, huntToolDetails[0].version, huntToolDetails[0]);
                 }
             });
+        }
+
+        function _loadConnectorDetails(connectorName, connectorVersion, sourceControl) {
```

Comment:
We should move this code to *service.js

Reply:
Its only used once so not created service